internal/driver/glfw/window.go

```diff
@@ -74,6 +74,7 @@ type window struct {
 	mouseLastClick     fyne.CanvasObject
 	mousePressed       fyne.CanvasObject
 	onClosed           func()
+	willClose          func()
```

Comment:
IMO more intuitive would be to have `willClose func() bool`, i.e. the handler decides if the window is closed.

Reply:
that was the initial suggestion and it was replaced for simplification so user should close window explicitly in the new handler

Reply:
Not very intuitive IMHO. But I'm OK with that as well. I would only change the name of the handler from `willClose` to `onCloseTry` or something similar.

Reply:
perhaps SetCloseIntercept()? that almost implies that follow-on action is required if set...